packages/emotion-primitives/src/createEmotion.js

```diff
@@ -0,0 +1,113 @@
+import * as React from 'react'
+import reactPrimitives from 'react-primitives'
+import PropTypes from 'prop-types'
+
+import { getStyles } from './getStyles'
+import { convertToRNStyles } from './convertToRNStyles'
+
+const primitives = ['Text', 'View', 'Image']
+
+const isValidPrimitive = primitive => primitives.indexOf(primitive) > -1
+
+const getPrimitive = primitive => {
+  if (typeof primitive === 'string' && isValidPrimitive(primitive)) {
+    return reactPrimitives[primitive]
+  } else if (typeof primitive === 'string' && !isValidPrimitive(primitive)) {
+    throw new Error(
+      `Cannot style invalid primitive ${primitive}. Expected primitive to be one of ['Text', 'View', 'Image']`
+    )
+  } else if (typeof primitive === 'function') {
+    return primitive
+  }
+}
+
+function evalStyles(context, Comp, styles, styleOverrides) {
+  // Assign static property so that the styles can be reused (like in withComponent)
+  Comp.styles = convertToRNStyles.call(context, styles)
```

Comment:
This seems to be called on each render so the styles assigned as statics on the component will change depending on the props. Maybe the styles array created at the start of `createStyledComponent` should be assigned as a static on the component instead?

Reply:
We can do that but, using this has an advantage. We can use reuse the converted styles like -

```js
const Text = emotion.text`color: red; padding: 10px;`

// After calling convertToRNStyles, styles get converted to RN
// like this -> [73]

// If we decide to assign the styles array then we won't be able to
// reuse the styles on another component because styles array
// would be like this [['color: red; padding: 10px;', raw: [...]]].

const Author = emotion.text`${Text.styles} position: absolute; right: 0;` 
```

Reply:
This is good when composing the styles like using `css` on Web.